creator/nx_creator_ptycho.py

```diff
@@ -117,44 +118,43 @@ def _create_dataset(self,
 
     def _check_unit(self, name, expected, supplied):
         """
-        Unit check for supplied units
+        Return `True` if conversion is possible between expected and supplied units.
 
-        Our test for units should check if the supplied
-        units string can be mapped into the expected units for that field.
         If arbitrary units are supplied in form of 'au', 'a.u.' or 'a.u' no conversion is applied
-        and pint if not used for the unit check.
-
+        and pint is not used for the unit check.
         :param : name of field
         :param : expected units
         :param : units string that was given
-        :return *bool*: `True` if units conversion is possible:
+        :return *bool*: `True` if units conversion is possible
         """
 
         # catch arbitrary unit separately from pint --> point that out in documentation
         if supplied in ['au', 'a.u.', 'a.u']:
-            logger.info("Info: arbitrary units supplied for '%s' in form of '%s' np unit conversion applicable",
+            logger.info("Arbitrary units supplied for '%s' in form of '%s' no unit conversion or "
+                        "pint unit check applicable",
                         name,
                         supplied)
             return True
         else:
             ureg = pint.UnitRegistry()
             user = 1.0 * ureg(supplied)
             try:
-                user.to(expected)
+
+                user.check(expected)
                 return True
             except pint.DimensionalityError:
-                logger.warning("WARNING: '%s': Supplied unit (%s) does not match expected units (%s)",
-                               name,
-                               supplied,
-                               expected)
+                warnings.warn("WARNING: '%s': Supplied unit (%s) does not match expected units (%s)",
```

Comment:
`warnings.warn` Already includes the word `UserWarning` in the output, so it's probably not necessary to include it again.
```suggestion
                warnings.warn("'%s': Supplied unit (%s) does not match expected units (%s)",
```